cmd/authgear/server/server.go

```diff
@@ -73,67 +63,31 @@ func (c *Controller) Start() {
 		c.logger.WithError(err).Fatal("cannot open configuration")
 	}
 
-	c.wg = new(sync.WaitGroup)
-	shutdown := make(chan struct{})
-	c.shutdown = shutdown
+	var specs []server.Spec
 
 	if c.ServeMain {
-		c.startServer(cfg, serverMain, cfg.ListenAddr, auth.NewRouter(p, configSource))
-	}
-	if c.ServeResolver {
-		c.startServer(cfg, serverResolver, cfg.ResolverListenAddr, resolver.NewRouter(p, configSource))
-	}
-	if c.ServeAdmin {
-		c.startServer(cfg, serverAdminAPI, cfg.AdminListenAddr, admin.NewRouter(p, configSource))
+		specs = append(specs, server.Spec{
+			Name:          string(serverMain),
```

Comment:
I suppose the `serverType` type is no longer needed after this refactor. Let's remove it and use string literal as server name directly.